log/tokenizer_test.go

```diff
@@ -7,41 +7,78 @@ import (
 )
 
 func TestTokenizer(t *testing.T) {
-	tokens, err := tokenize("loki=something,s.e=2231,s=12,12=3,a=[1,2,3],b=[1],s=c")
-	assert.Equal(t, []token{
+	tests := []struct {
+		name       string
+		input      string
+		wantTokens []token
+	}{
 		{
-			key:   "loki",
-			value: "something",
-		},
-		{
-			key:   "s.e",
-			value: "2231",
-		},
-		{
-			key:   "s",
-			value: "12",
+			name:  "good keys",
+			input: "loki=something,s.e=2231,s=12,12=3,a=[1,2,3],b=[1],s=c",
+			wantTokens: []token{
+				{key: "loki", value: "something"},
+				{key: "s.e", value: "2231"},
+				{key: "s", value: "12"},
+				{key: "12", value: "3"},
+				{key: "a", value: "1,2,3", inside: '['},
+				{key: "b", value: "1", inside: '['},
+				{key: "s", value: "c"},
+			},
 		},
+	}
+
+	for _, test := range tests {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+
+			gotTokens, err := tokenize(test.input)
+			if err != nil {
+				t.Fatal("didn't expect error")
+			}
+			assert.Equal(t, test.wantTokens, gotTokens)
+		})
+	}
+}
```

Comment:
It doesn't seems we need a table test here so I would prefer a less-nested solution. 

```go
func TestTokenizer(t *testing.T) {
	t.Parallel()

	input := "loki=something,s.e=2231,s=12,12=3,a=[1,2,3],b=[1],s=c"
	tokens, err := tokenize(input)
	require.NoError(t, err)

	expected := []token{
		{key: "loki", value: "something"},
		{key: "s.e", value: "2231"},
		{key: "s", value: "12"},
		{key: "12", value: "3"},
		{key: "a", value: "1,2,3", inside: '['},
		{key: "b", value: "1", inside: '['},
		{key: "s", value: "c"},
	}
	assert.Equal(t, expected, tokens)
}
```